daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/ContractDiscriminatorFreshnessCheckSpec.scala

```diff
@@ -354,14 +354,16 @@ class ContractDiscriminatorFreshnessCheckSpec
         pcs = _ => None,
         keys = _ => None,
       )
-      engine.replay(
-        submitters = Set(alice),
-        tx,
-        ledgerEffectiveTime = txMeta.submissionTime,
-        participantId = participant,
-        submissionTime = txMeta.submissionTime,
-        submissionSeed = txMeta.submissionSeed.get,
-      ) shouldBe a[ResultDone[_]]
+      engine
```

Comment:
@remyhaemmerle-da side note: This is a case where we do not serialize/deserialize and probably tons of other tests as well. So definitely in favor of doing something like your suggestion of normalizing in `finish` (while figuring out what we do with canton & scenarios).